### ISSUE TITLE
Bug in census_widendeep.py downloading of test data

### ISSUE THREAD

**Issue by ancadumitrache** [tensorflow/tensorflow#9863]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: no
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: nightly builds
- **TensorFlow version (use command below)**: 1.1.0-rc2

### Describe the problem
There seems to be an error in the downloading of the test data in the distributed training example `census_widendeep.py`. The test data is loaded from the file before the file is actually downloaded (lines 152-153):

```python
test_file = open(test_file_path)
urllib.urlretrieve(test_data_url, test_file_path)
```

**Comment by girving** [comment]
@caisq That does seem odd.  A slightly larger snippet of code is

    # Retrieve data from disk (if available) or download from the web.
    train_file_path = os.path.join(data_dir, "adult.data")
    if os.path.isfile(train_file_path):
      print("Loading training data from file: %s" % train_file_path)
      train_file = open(train_file_path)
    else:
      urllib.urlretrieve(train_data_url, train_file_path)

    test_file_path = os.path.join(data_dir, "adult.test")
    if os.path.isfile(test_file_path):
      print("Loading test data from file: %s" % test_file_path)
      test_file = open(test_file_path)
    else:
      test_file = open(test_file_path)
      urllib.urlretrieve(test_data_url, test_file_path)

Do you know how this is supposed to work?

**Comment by itsmeolivia** [comment]
friendly ping: are you still having issues @ancadumitrache

**Comment by ancadumitrache** [comment]
@itsmeolivia It worked after I shuffled the URL retrieve and file open commands (for both train and test data):

```
# Retrieve data from disk (if available) or download from the web.
train_file_path = os.path.join(data_dir, "adult.data")
if os.path.isfile(train_file_path):
  print("Loading training data from file: %s" % train_file_path)
  train_file = open(train_file_path)
else:
  urllib.urlretrieve(train_data_url, train_file_path)
  train_file = open(train_file_path)

test_file_path = os.path.join(data_dir, "adult.test")
if os.path.isfile(test_file_path):
  print("Loading test data from file: %s" % test_file_path)
  test_file = open(test_file_path)
else:
  urllib.urlretrieve(test_data_url, test_file_path)
  test_file = open(test_file_path)
```
